kiliautoml/utils/pytorchvision/trainer.py

```diff
@@ -25,19 +25,20 @@ def train_model_pytorch(
     since = time.time()
 
     device = torch.device("cuda:0" if torch.cuda.is_available() else "cpu")
-
+    if verbose >= 2:
+        print("Start model training on device: {}".format(device))
     criterion = nn.CrossEntropyLoss()
     optimizer = optim.SGD(model.parameters(), lr=0.001, momentum=0.9)
 
     model = model.to(device)
-    dataset_sizes = {x: len(dataloaders[x]) for x in ["train", "val"]}
+    dataset_sizes = {x: len(dataloaders[x].dataset) for x in ["train", "val"]}
 
     # Decay LR by a factor of 0.1 every 7 epochs
     scheduler = lr_scheduler.StepLR(optimizer, step_size=7, gamma=0.1)
 
     best_model_wts = copy.deepcopy(model.state_dict())
     best_acc = 0.0
-
+    best_loss = float("inf")
     for epoch in range(epochs):
```

Comment:
add tqdm here
from tqdm.autonotebook import tqdm

Reply:
trange(epochs, desc='Training')

Reply:
OK, but do I keep the             
```
print(f"Epoch {epoch + 1}/{epochs}")
print("-" * 10)
```
then ?

Reply:
Ok, after tries, I put 
```
    for _ in trange(epochs, desc="Training - Epoch"):
        if verbose >= 2:
            print("-" * 10)

```